notice.go

```diff
@@ -77,6 +78,13 @@ func NewNotice(e interface{}, req *http.Request, depth int) *Notice {
 		notice.Context[k] = v
 	}
 
+	switch n := e.(type) {
+	case *Notice:
+		if s := n.Context["severity"]; s != "" {
+			notice.Context["severity"] = s
+		}
+	}
```

Comment:
You need to add something like this to NewNotice

```
notice, ok := e.(*Notice)
if ok {
    return notice
}
```

Even then it looks like a hack, but not a bad one.